### ISSUE TITLE
Items - 'Show Details' button fails to tell screen readers if it has been toggled

### ISSUE THREAD

**Issue by aviannamiller** [omeka/Omeka#1123]
"Description: On the 'Items' page, there is a button labeled 'Show Details'. If screen reader users select this button with 'Enter' keystroke, the user will not be told by the screen reader that the button has been toggled on to show the details of all the items. This can be confusing for screen reader users because they may question if the button toggled on or off.

Note: The 'Details' links on each item also fails to give a status message if users decide to individually expand them.

Note: The button is not a true toggle button since the label changes when the button is pressed. So adding the aria-pressed attribute would not be a sufficient fix for this issue.  https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-pressed

Expected Results: When pressing the button to change the state of the page, screen reader users are informed of the state change and informed of the change in the button label. Users should clearly understand whether item details are now visible or hidden.

The switch pattern is recommended: https://www.w3.org/WAI/ARIA/apg/patterns/switch/

Actual Results: The screen reader provides no indication that the button has changed state after activation. Users receive no feedback confirming whether the details view has been toggled on or off."

4.1.2 Name, Role, Value; Level A; WCAG 2.0
4.1.3 Status Messages; Level AA; WCAG 2.1

**Comment by kimisgold** [comment]
A revised pattern for the show details interactions is now available on [`a11y/show-details-toggle`](https://github.com/omeka/Omeka/tree/show-details-toggle).

* "Details" no longer presents as a link inline with "Edit" and "Delete" and has been converted to a button
* Each button should have `aria-expanded` and `aria-controls` attributes
* The "Show/Hide Details" label has been updated to "Show/Hide All Details"
* There are now screen reader alerts for the "Show/Hide All Details" buttons

**Comment by aviannamiller** [comment]
All set on the "Show/Hide All Details" aspect, that all checks out.

For the revised toggle buttons, can the name of each item be referred to in the readout along with "Details"? Perhaps this is in the territory of https://github.com/omeka/Omeka/issues/1117 and has already been handled there, but I haven't yet taken a look at those changes.

I'm wondering about the focus order now. After expanding the toggle, focus does not move immediately into the details container. If the button has now relocated to the item name, perhaps the details container needs to instead appear next in the sequence, above the Edit and Delete links? What do you think?

In terms of the button appearance, should there remain a visible "Details" label or something like that? In some ways, toggle purpose feels a bit unclear, especially on mobile:

<img width="520" height="323" alt="Detail of items in browse items table. Each item has the item name as a link, followed by a right facing toggle on the next line, then a Featured or Private label, and finally, Edit and Delete links." src="https://github.com/user-attachments/assets/5def51a8-d796-4f1f-8f48-6c8278d2b329" />

Speaking of mobile, do we need to lose the button appearance of the Edit and Delete links in narrower views?

**Comment by aviannamiller** [comment]
Realized #1117 as referenced prior is not totally relevant here, but we should be looking to fine-tune grouping of associated elements within table cells of browse views, if we don't already have an issue for that.
